src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperDescriptorConventions.cs

```diff
@@ -0,0 +1,99 @@
+                && descriptor.PropertyBag.ContainsKey(ViewComponentTagHelperProperty));
+        }
+
+        /// <summary>
+        /// Retrieves the view component name from the tag helper property bag.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The short name of the <see cref="ViewComponent"/> represented by the <see cref="TagHelperDescriptor"/>,
+        /// or null if the tag helper does not represent a view component.</returns>
+        public static string GetViewComponentName(TagHelperDescriptor descriptor)
+        {
+            if (!IsViewComponentDescriptor(descriptor)) return null;
+
+            var viewComponentName = descriptor.PropertyBag[ViewComponentProperty];
+            return viewComponentName;
+        }
+
+        /// <summary>
+        /// Retrieves the view component tag helper name from the tag helper property bag.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The custom type name for a <see cref="ViewComponent"/> represented by the <see cref="TagHelperDescriptor"/>,
+        /// or null if the tag helper does not represent a view component.</returns>
+        public static string GetViewComponentTagHelperName(TagHelperDescriptor descriptor)
+        {
+            if (!IsViewComponentDescriptor(descriptor)) return null;
+
+            var viewComponentTagHelperName = descriptor.PropertyBag[ViewComponentTagHelperProperty];
+            return viewComponentTagHelperName;
+        }
+
+        /// <summary>
+        /// Retrieves the assembly name from a view component descriptor.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="ViewComponentDescriptor"/>.</param>
+        /// <returns>The name of the assembly containing the <see cref="ViewComponentDescriptor"/>.</returns>
+        public static string GetAssemblyName(ViewComponentDescriptor descriptor) =>
+            descriptor.TypeInfo.Assembly.GetName().Name;
+
+        /// <summary>
+        /// Creates a custom tag name from a view component descriptor.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="ViewComponentDescriptor"/>.</param>
+        /// <returns>The tag name used to invoke a <see cref="ViewComponent"/> as a tag helper.</returns>
+        public static string GetTagName(ViewComponentDescriptor descriptor) =>
+            $"vc:{TagHelperDescriptorFactory.ToHtmlCase(descriptor.ShortName)}";
+
+        /// <summary>
+        /// Creates a custom type name from a view component descriptor.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="ViewComponentDescriptor"/>.</param>
+        /// <returns>A custom type name for a tag helper representing a view component.</returns>
+        public static string GetTypeName(ViewComponentDescriptor descriptor) =>
```

Comment:
Generally don't use the lambda syntax when the code gets past a single statement. Just use a regular method body here.